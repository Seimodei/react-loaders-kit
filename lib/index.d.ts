import RotatingBoxesLoader from './rotatingBoxes/RotatingBoxesLoader';
import AlternatingOrbitsLoader from './alternatingOrbits/AlternatingOrbitsLoader';
import SpinningCircleLoader from './spinningCircle/SpinningCircleLoader';
import SpinningOrbitLoader from './spinningOrbit/SpinningOrbitLoader';
import PulseBubbleLoader from './pulseBubble/PulseBubbleLoader';
import BouncyBallsLoader from './bouncyBalls/BouncyBallsLoader';
import DyingLightLoader from './dyingLight/DyingLightLoader';
import CirclePopLoader from './circlePop/CirclePopLoader';
import GooeyCircleLoader from './gooeyCircle/GooeyCircleLoader';
import LiquidLoader from './liquid/LiquidLoader';
import BlurrySquareLoader from './blurrySquare/BlurrySquareLoader';
import JellyBounceLoader from './jellyBounce/JellyBounceLoader';
import AtomLoader from './atom/AtomLoader';
import ElasticCircleLoader from './elasticCircle/ElasticCircleLoader';
import FlipLoader from './flip/FlipLoader';
import MovingSquareLoader from './movingSquare/MovingSquareLoader';
import FlippingCubeLoader from './flippingCube/FlippingCubeLoader';
import SlidingCubeLoader from './slidingCube/SlidingCubeLoader';
import SwingingCubeLoader from './swingingCube/SwingingCubeLoader';
import SwitchingCubeLoader from './switchingCube/SwitchingCubeLoader';
import LinneardLoader from './linneard/LinneardLoader';
import FillCircleLoader from './fillCircle/FillCircleLoader';
import GooeyLoader1 from './gooey1/GooeyLoader1';
import GooeyLoader2 from './gooey2/GooeyLoader2';
import HelixLoader from './helix/HelixLoader';
import VolumeLoader from './volume/VolumeLoader';
import VortexLoader from './vortex/VortexLoader';
import SlicesLoader from './slices/SlicesLoader';
import SphereLoader from './sphere/SphereLoader';
import BarsLoader from './bars/BarsLoader';
import ClockLoader from './clock/ClockLoader';
import WaveLoader from './wave/WaveLoader';
import TextureLoader from './texture/TextureLoader';
import BatteryLoader from './battery/BatteryLoader';
import HydrogenLoader from './hydrogen/HydrogenLoader';
import FillSquareLoader from './fillSquare/FillSquareLoader';
import HypnosisLoader from './hypnosis/HypnosisLoader';
import LineLoader from './line/LineLoader';
import CircleFadeLoader from './circleFade/CircleFadeLoader';
import EightBitLoader from './eightBit/EightBitLoader';
import PingPongLoader from './pingPong/PingPongLoader';
import KissyBallsLoader from './kissyBalls/KissyBallsLoader';
import DrawWaveLoader from './drawWave/DrawWaveLoader';
import BlobLoader from './blob/BlobLoader';
import BarsLoader2 from './bars2/BarsLoader2';
import RotatingCircleLoader from './rotatingCircle/RotatingCircleLoader';
import NotepadLoader from './notepad/NotepadLoader';
import DoubleSquareLoader from './doubleSquare/DoubleSquareLoader';
import BrokenCirclesLoader from './brokenCircles/BrokenCirclesLoader';
import PipLoader from './pip/PipLoader';
import DotsLoader from './dots/DotsLoader';
export { RotatingBoxesLoader, AlternatingOrbitsLoader, SpinningCircleLoader, PulseBubbleLoader, SpinningOrbitLoader, BouncyBallsLoader, DyingLightLoader, CirclePopLoader, GooeyCircleLoader, LiquidLoader, BlurrySquareLoader, JellyBounceLoader, AtomLoader, ElasticCircleLoader, FlipLoader, MovingSquareLoader, FlippingCubeLoader, SlidingCubeLoader, SwingingCubeLoader, SwitchingCubeLoader, LinneardLoader, FillCircleLoader, GooeyLoader1, GooeyLoader2, SphereLoader, SlicesLoader, BatteryLoader, HydrogenLoader, HelixLoader, VolumeLoader, VortexLoader, BarsLoader, ClockLoader, WaveLoader, TextureLoader, FillSquareLoader, HypnosisLoader, LineLoader, CircleFadeLoader, EightBitLoader, PingPongLoader, KissyBallsLoader, DrawWaveLoader, BlobLoader, BarsLoader2, RotatingCircleLoader, NotepadLoader, DoubleSquareLoader, BrokenCirclesLoader, PipLoader, DotsLoader };
