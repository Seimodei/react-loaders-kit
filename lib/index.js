"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var RotatingBoxesLoader_1 = __importDefault(require("./rotatingBoxes/RotatingBoxesLoader"));
exports.RotatingBoxesLoader = RotatingBoxesLoader_1.default;
var AlternatingOrbitsLoader_1 = __importDefault(require("./alternatingOrbits/AlternatingOrbitsLoader"));
exports.AlternatingOrbitsLoader = AlternatingOrbitsLoader_1.default;
var SpinningCircleLoader_1 = __importDefault(require("./spinningCircle/SpinningCircleLoader"));
exports.SpinningCircleLoader = SpinningCircleLoader_1.default;
var SpinningOrbitLoader_1 = __importDefault(require("./spinningOrbit/SpinningOrbitLoader"));
exports.SpinningOrbitLoader = SpinningOrbitLoader_1.default;
var PulseBubbleLoader_1 = __importDefault(require("./pulseBubble/PulseBubbleLoader"));
exports.PulseBubbleLoader = PulseBubbleLoader_1.default;
var BouncyBallsLoader_1 = __importDefault(require("./bouncyBalls/BouncyBallsLoader"));
exports.BouncyBallsLoader = BouncyBallsLoader_1.default;
var DyingLightLoader_1 = __importDefault(require("./dyingLight/DyingLightLoader"));
exports.DyingLightLoader = DyingLightLoader_1.default;
var CirclePopLoader_1 = __importDefault(require("./circlePop/CirclePopLoader"));
exports.CirclePopLoader = CirclePopLoader_1.default;
var GooeyCircleLoader_1 = __importDefault(require("./gooeyCircle/GooeyCircleLoader"));
exports.GooeyCircleLoader = GooeyCircleLoader_1.default;
var LiquidLoader_1 = __importDefault(require("./liquid/LiquidLoader"));
exports.LiquidLoader = LiquidLoader_1.default;
var BlurrySquareLoader_1 = __importDefault(require("./blurrySquare/BlurrySquareLoader"));
exports.BlurrySquareLoader = BlurrySquareLoader_1.default;
var JellyBounceLoader_1 = __importDefault(require("./jellyBounce/JellyBounceLoader"));
exports.JellyBounceLoader = JellyBounceLoader_1.default;
var AtomLoader_1 = __importDefault(require("./atom/AtomLoader"));
exports.AtomLoader = AtomLoader_1.default;
var ElasticCircleLoader_1 = __importDefault(require("./elasticCircle/ElasticCircleLoader"));
exports.ElasticCircleLoader = ElasticCircleLoader_1.default;
var FlipLoader_1 = __importDefault(require("./flip/FlipLoader"));
exports.FlipLoader = FlipLoader_1.default;
var MovingSquareLoader_1 = __importDefault(require("./movingSquare/MovingSquareLoader"));
exports.MovingSquareLoader = MovingSquareLoader_1.default;
var FlippingCubeLoader_1 = __importDefault(require("./flippingCube/FlippingCubeLoader"));
exports.FlippingCubeLoader = FlippingCubeLoader_1.default;
var SlidingCubeLoader_1 = __importDefault(require("./slidingCube/SlidingCubeLoader"));
exports.SlidingCubeLoader = SlidingCubeLoader_1.default;
var SwingingCubeLoader_1 = __importDefault(require("./swingingCube/SwingingCubeLoader"));
exports.SwingingCubeLoader = SwingingCubeLoader_1.default;
var SwitchingCubeLoader_1 = __importDefault(require("./switchingCube/SwitchingCubeLoader"));
exports.SwitchingCubeLoader = SwitchingCubeLoader_1.default;
var LinneardLoader_1 = __importDefault(require("./linneard/LinneardLoader"));
exports.LinneardLoader = LinneardLoader_1.default;
var FillCircleLoader_1 = __importDefault(require("./fillCircle/FillCircleLoader"));
exports.FillCircleLoader = FillCircleLoader_1.default;
var GooeyLoader1_1 = __importDefault(require("./gooey1/GooeyLoader1"));
exports.GooeyLoader1 = GooeyLoader1_1.default;
var GooeyLoader2_1 = __importDefault(require("./gooey2/GooeyLoader2"));
exports.GooeyLoader2 = GooeyLoader2_1.default;
var HelixLoader_1 = __importDefault(require("./helix/HelixLoader"));
exports.HelixLoader = HelixLoader_1.default;
var VolumeLoader_1 = __importDefault(require("./volume/VolumeLoader"));
exports.VolumeLoader = VolumeLoader_1.default;
var VortexLoader_1 = __importDefault(require("./vortex/VortexLoader"));
exports.VortexLoader = VortexLoader_1.default;
var SlicesLoader_1 = __importDefault(require("./slices/SlicesLoader"));
exports.SlicesLoader = SlicesLoader_1.default;
var SphereLoader_1 = __importDefault(require("./sphere/SphereLoader"));
exports.SphereLoader = SphereLoader_1.default;
var BarsLoader_1 = __importDefault(require("./bars/BarsLoader"));
exports.BarsLoader = BarsLoader_1.default;
var ClockLoader_1 = __importDefault(require("./clock/ClockLoader"));
exports.ClockLoader = ClockLoader_1.default;
var WaveLoader_1 = __importDefault(require("./wave/WaveLoader"));
exports.WaveLoader = WaveLoader_1.default;
var TextureLoader_1 = __importDefault(require("./texture/TextureLoader"));
exports.TextureLoader = TextureLoader_1.default;
var BatteryLoader_1 = __importDefault(require("./battery/BatteryLoader"));
exports.BatteryLoader = BatteryLoader_1.default;
var HydrogenLoader_1 = __importDefault(require("./hydrogen/HydrogenLoader"));
exports.HydrogenLoader = HydrogenLoader_1.default;
var FillSquareLoader_1 = __importDefault(require("./fillSquare/FillSquareLoader"));
exports.FillSquareLoader = FillSquareLoader_1.default;
var HypnosisLoader_1 = __importDefault(require("./hypnosis/HypnosisLoader"));
exports.HypnosisLoader = HypnosisLoader_1.default;
var LineLoader_1 = __importDefault(require("./line/LineLoader"));
exports.LineLoader = LineLoader_1.default;
var CircleFadeLoader_1 = __importDefault(require("./circleFade/CircleFadeLoader"));
exports.CircleFadeLoader = CircleFadeLoader_1.default;
var EightBitLoader_1 = __importDefault(require("./eightBit/EightBitLoader"));
exports.EightBitLoader = EightBitLoader_1.default;
var PingPongLoader_1 = __importDefault(require("./pingPong/PingPongLoader"));
exports.PingPongLoader = PingPongLoader_1.default;
var KissyBallsLoader_1 = __importDefault(require("./kissyBalls/KissyBallsLoader"));
exports.KissyBallsLoader = KissyBallsLoader_1.default;
var DrawWaveLoader_1 = __importDefault(require("./drawWave/DrawWaveLoader"));
exports.DrawWaveLoader = DrawWaveLoader_1.default;
var BlobLoader_1 = __importDefault(require("./blob/BlobLoader"));
exports.BlobLoader = BlobLoader_1.default;
var BarsLoader2_1 = __importDefault(require("./bars2/BarsLoader2"));
exports.BarsLoader2 = BarsLoader2_1.default;
var RotatingCircleLoader_1 = __importDefault(require("./rotatingCircle/RotatingCircleLoader"));
exports.RotatingCircleLoader = RotatingCircleLoader_1.default;
var NotepadLoader_1 = __importDefault(require("./notepad/NotepadLoader"));
exports.NotepadLoader = NotepadLoader_1.default;
var DoubleSquareLoader_1 = __importDefault(require("./doubleSquare/DoubleSquareLoader"));
exports.DoubleSquareLoader = DoubleSquareLoader_1.default;
var BrokenCirclesLoader_1 = __importDefault(require("./brokenCircles/BrokenCirclesLoader"));
exports.BrokenCirclesLoader = BrokenCirclesLoader_1.default;
var PipLoader_1 = __importDefault(require("./pip/PipLoader"));
exports.PipLoader = PipLoader_1.default;
var DotsLoader_1 = __importDefault(require("./dots/DotsLoader"));
exports.DotsLoader = DotsLoader_1.default;
