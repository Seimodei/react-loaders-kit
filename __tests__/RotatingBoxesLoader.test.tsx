import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { RotatingBoxesLoader } from '../lib/index';

configure({ adapter: new Adapter() });


describe('RotatingBoxesLoader', () => {
    describe('default', () => {
        const wrapper = mount(<RotatingBoxesLoader loading={true} />);
        const props = wrapper.props();

        it('should match snapshot', () => {
            expect(wrapper.find(RotatingBoxesLoader)).toMatchSnapshot();
        })

        it('should render with default loading true', () => {
            expect(props.loading).toEqual(true);
        })
    });
});

