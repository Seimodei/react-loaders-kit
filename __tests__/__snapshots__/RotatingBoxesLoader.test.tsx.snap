// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotatingBoxesLoader default should match snapshot 1`] = `
<RotatingBoxesLoader
  loading={true}
>
  <Styled(div)>
    <div
      className="css-lo26e5"
    >
      <Styled(div)>
        <div
          className="css-1olnkr2"
        >
          <Styled(div)
            style={
              Object {
                "backgroundColor": "#fafafa",
              }
            }
          >
            <div
              className="css-s08p0c"
              style={
                Object {
                  "backgroundColor": "#fafafa",
                }
              }
            />
          </Styled(div)>
        </div>
      </Styled(div)>
      <Styled(div)>
        <div
          className="css-e5pv52"
        >
          <Styled(div)
            style={
              Object {
                "backgroundColor": "#fafafa",
              }
            }
          >
            <div
              className="css-s08p0c"
              style={
                Object {
                  "backgroundColor": "#fafafa",
                }
              }
            />
          </Styled(div)>
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</RotatingBoxesLoader>
`;
